### PR TITLE
Support canceling cross-clusters search requests

### PR DESCRIPTION
**Pull request by dnhatn**
This commit supports canceling cross-clusters search requests. Several important changes in this commit:

- Set the parent task for [CCS search requests](https://github.com/elastic/elasticsearch/pull/66206/files#diff-24921eca27bf57c7b3d8d3ba7f02acbf6449d91e43e49bbd9df6377843adb19aL372)
- Keep track of underlying connections instead of [proxy connections](https://github.com/elastic/elasticsearch/pull/66206/files#diff-27dc81a65a8d02c5e6ea7e5a86e91659cb8b48c9aed0faa3fc77cd3c73cfdaebR620)
- Assign the parent task for [proxy requests](https://github.com/elastic/elasticsearch/pull/66206/files#diff-f528d84b397dcef0aedf423793cc76ab1b0311b15655844e825ac9c1d21932c4R124)

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/action/search/TransportSearchIT.java [hunk 1]
```diff
@@ -159,6 +161,7 @@ public void testLocalClusterAlias() {
     }
 
     public void testAbsoluteStartMillis() {
+        TaskId takeId = new TaskId("node", randomNonNegativeLong());
```
Comment (jimczi) [inline server/src/internalClusterTest/java/org/elasticsearch/action/search/TransportSearchIT.java:164]: `parentTaskId` ?

File: server/src/internalClusterTest/java/org/elasticsearch/action/search/TransportSearchIT.java [hunk 2]
```diff
@@ -131,9 +132,10 @@ public void testLocalClusterAlias() {
         indexRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL);
         IndexResponse indexResponse = client().index(indexRequest).actionGet();
         assertEquals(RestStatus.CREATED, indexResponse.status());
+        TaskId taskId = new TaskId("node", randomNonNegativeLong());
```
Comment (jimczi) [inline server/src/internalClusterTest/java/org/elasticsearch/action/search/TransportSearchIT.java:135]: `parentTaskId` ?

**Review by jimczi — APPROVED** [review]
Nicely done, thanks @dnhatn !

File: server/src/internalClusterTest/java/org/elasticsearch/action/search/TransportSearchIT.java (hunk 1 above)
Comment (dnhatn) [reply server/src/internalClusterTest/java/org/elasticsearch/action/search/TransportSearchIT.java]: Addressed in https://github.com/elastic/elasticsearch/pull/66206/commits/646aeab9ed596e00360112b72e628839aa98a47c

File: server/src/internalClusterTest/java/org/elasticsearch/action/search/TransportSearchIT.java (hunk 2 above)
Comment (dnhatn) [reply server/src/internalClusterTest/java/org/elasticsearch/action/search/TransportSearchIT.java]: Addressed in https://github.com/elastic/elasticsearch/pull/66206/commits/646aeab9ed596e00360112b72e628839aa98a47c

**Comment by dnhatn** [comment]
Thanks Jim!

**Pull request merged**